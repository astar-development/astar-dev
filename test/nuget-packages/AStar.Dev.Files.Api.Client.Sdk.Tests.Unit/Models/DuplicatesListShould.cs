using AStar.Dev.Utilities;

namespace AStar.Dev.Files.Api.Client.Sdk.Models;

[TestSubject(typeof(DuplicatesList))]
public class DuplicatesListShould
{
    [Fact]
    public void ContainTheExpectedProperties()
        => new DuplicatesList(new(123456, 987, 654),
                              [
                                  new(1, 2, "MockFileName", "MockDirectoryName", 12, 345, 543, "MockFileHandle", true, 32, DateTimeOffset.MaxValue, DateTimeOffset.MaxValue.AddDays(-1),
                                      true, true, true, true)
                              ])
           .ToJson()
           .ShouldMatchApproved();
}
