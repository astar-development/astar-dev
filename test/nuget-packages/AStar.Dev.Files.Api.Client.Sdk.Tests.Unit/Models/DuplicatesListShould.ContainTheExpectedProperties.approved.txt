{
  "fileSize": {
    "fileSize": 123456,
    "height": 987,
    "width": 654,
    "fileSizeForDisplay": "120.56 Kb"
  },
  "duplicates": [
    {
      "id": 1,
      "fileAccessDetailId": 2,
      "fileName": "MockFileName",
      "directoryName": "MockDirectoryName",
      "height": 12,
      "width": 345,
      "fileSize": 543,
      "fileHandle": "MockFileHandle",
      "isImage": true,
      "instances": 32,
      "detailsLastUpdated": "9999-12-31T23:59:59.9999999+00:00",
      "lastViewed": "9999-12-30T23:59:59.9999999+00:00",
      "softDeleted": true,
      "softDeletePending": true,
      "moveRequired": true,
      "hardDeletePending": true,
      "fullNameWithPath": "MockDirectoryName/MockFileName"
    }
  ]
}